helix-term/src/keymap.rs

```diff
@@ -62,247 +154,300 @@ impl DerefMut for Keymaps {
 
 impl Default for Keymaps {
     fn default() -> Keymaps {
-        let normal = hashmap!(
-            key!('h') => Command::move_char_left,
-            key!('j') => Command::move_line_down,
-            key!('k') => Command::move_line_up,
-            key!('l') => Command::move_char_right,
-
-            key!(Left) => Command::move_char_left,
-            key!(Down) => Command::move_line_down,
-            key!(Up) => Command::move_line_up,
-            key!(Right) => Command::move_char_right,
-
-            key!('t') => Command::find_till_char,
-            key!('f') => Command::find_next_char,
-            key!('T') => Command::till_prev_char,
-            key!('F') => Command::find_prev_char,
-            // and matching set for select mode (extend)
-            //
-            key!('r') => Command::replace,
-            key!('R') => Command::replace_with_yanked,
-
-            key!(Home) => Command::goto_line_start,
-            key!(End) => Command::goto_line_end,
-
-            key!('w') => Command::move_next_word_start,
-            key!('b') => Command::move_prev_word_start,
-            key!('e') => Command::move_next_word_end,
-
-            key!('W') => Command::move_next_long_word_start,
-            key!('B') => Command::move_prev_long_word_start,
-            key!('E') => Command::move_next_long_word_end,
-
-            key!('v') => Command::select_mode,
-            key!('g') => Command::goto_mode,
-            key!(':') => Command::command_mode,
-
-            key!('i') => Command::insert_mode,
-            key!('I') => Command::prepend_to_line,
-            key!('a') => Command::append_mode,
-            key!('A') => Command::append_to_line,
-            key!('o') => Command::open_below,
-            key!('O') => Command::open_above,
+        let normal = keymap!(
+            "h" => move_char_left,
+            "j" => move_line_down,
+            "k" => move_line_up,
+            "l" => move_char_right,
+
+            "left" => move_char_left,
+            "down" => move_line_down,
+            "up" => move_line_up,
+            "right" => move_char_right,
+
+            "t" => find_till_char,
+            "f" => find_next_char,
+            "T" => till_prev_char,
+            "F" => find_prev_char,
+            "r" => replace,
+            "R" => replace_with_yanked,
+
+            "home" => goto_line_start,
+            "end" => goto_line_end,
+
+            "w" => move_next_word_start,
+            "b" => move_prev_word_start,
+            "e" => move_next_word_end,
+
+            "W" => move_next_long_word_start,
+            "B" => move_prev_long_word_start,
+            "E" => move_next_long_word_end,
+
+            "v" => select_mode,
+            "g" => {
```

Comment:
Goto line number doesn't work now since there is no separate function for goto mode. Adding a function for goto mode only will require treating it as a special case especially for handling the infobox and user remappings. A solution is to add it to `gg` (keybind would change from `82g` to `82gg`, like in vim).

Reply:
Can't we add hooks?

Reply:
But that would make it impossible to have any other key under `g` to take a count, and the user might remap some function to it that takes a count.

Reply:
The keys under g from what I see all does not need a count.

Reply:
The default ones don't, I meant the case as a user remapping some other function that takes count to a key under `g`. I will probably do this myself with `ge` to be vim-like (go to end of previous word) and it would take a count.